src/lib/resource.js

```diff
@@ -1,74 +1,393 @@
+ */
+const getMetadataValuesFormHtml = async (url) => {
+  let result;
+  try {
+    // Make a GET request to the specified URL
+    const res = await axios.get(url, requestOptions);
+
+    // Extract metadata values from the HTML response
+    const metadata = metadataPatterns
+      .flatMap((pattern) => ((res && res.data && typeof res.data === 'string')
+        ? [...(res?.data?.matchAll(pattern) || null)]
+        : []))
+      .reduce((acc, meta) => {
+        if (Array.isArray(meta)) {
+          const key = meta[1].toLowerCase();
+          const value = meta[2].trim();
+          if (value && value !== '') {
+            if (acc[key]) {
+              if (Array.isArray(acc[key])) {
+                acc[key].push(value);
+              } else {
+                acc[key] = [acc[key], value];
+              }
+            } else {
+              acc[key] = value;
+            }
+          }
+        }
+        return acc;
+      }, {});
+
+    // Prepare the result object
+    result = {
+      metadata: (typeof metadata === 'object' && Object.keys(metadata).length !== 0)
+        ? metadata
+        : null,
+      statusCode: overrideStatusCodeOnAuthRequired(
+        res?.status,
+        commonAuthNames,
+        res?.request?.res?.responseUrl !== url && res?.request?.res?.responseUrl,
+        true,
+      ),
+      mimeType: res?.headers['content-type'],
+    };
+  } catch (error) {
+    if (error.response) {
+      // Log an error message when unable to collect metadata due to a response error
+      auditLogger.error(
+        `Resource Queue: Unable to collect metadata from page scrape for Resource (URL: ${url}), received status code of ${error.response.status}. Please make sure this is a valid address:`,
+        error,
+      );
+      result = {
+        statusCode: error.response.status,
+        mimeType: error?.response?.headers['content-type'],
+      };
+    } else {
+      // Log an error message when unable to collect metadata due to an unexpected error
+      auditLogger.error(
+        `Resource Queue: Unable to collect metadata from page scrape for Resource (URL: ${url}). Please make sure this is a valid address:`,
+        error,
+      );
+      throw error;
+    }
+  }
+
+  return result;
+};
+
+/**
+ * Retrieves metadata values for a given URL.
+ * @param {string} url - The URL to retrieve metadata from.
+ * @returns {Promise<{ title: string | null, statusCode: number }>} - The title and status
+ * code of the resource.
+ */
+const getMetadataValues = async (url) => {
+  let statusCode; // Variable to store the status code of the resource.
+  let metadata; // Variable to store the metadata of the resource.
+  let title = null; // Variable to store the title of the resource, initialized as null.
+  let mimeType; // Variable to store the MIME type of the resource.
+
+  try {
+    const [
+      fromJson,
+      fromHtml,
+    ] = await Promise.allSettled([

```

Comment:
why would we need to get both?

Reply:
@GarrettEHill do we want both here?

Reply:
Spoke with @GarrettEHill we get both here to maximize chances for success in retrieval.  We then combine the two results via spread. The HTML values will only pull in if they are not blank. So we shouldn't have to worry about a case of JSON title being overwritten by HTML.

Reply:
Double tapping seems to like it would be a great way to get blocked by a rate limiter. The concern is making two requests where one would do, not that one would overwrite the other. We should only fetch HTML if we don't have JSON.